### PR TITLE
Accept r-value references in convert_table_for_return():

### PR DESCRIPTION
**Pull request by mythrocks**
`cudf::jni::convert_table_for_return()` is usually used on tables returned from a libcudf API call.
It currently requires an l-value reference for its table argument. This necessitates parking the result of libcudf call in an avoidable temp variable.
This commit adds the option to use an r-value reference. This allows table expressions to be used directly, reducing clutter.

Note:
  1. The previous signature is retained, because not all call sites can use the r-value 
     interface cleanly. (E.g. when the libcudf call is complex.)
  2. The third argument (vector<unique_ptr<column>>) has been converted from l-ref to r-ref,
     so that an empty default can be introduced.

This commit also includes minor code cleanup in the periphery of calls to `convert_table_for_return()`.

### CODE REVIEW

File: java/src/main/native/src/TableJni.cpp [hunk 1]
```diff
@@ -2997,8 +2997,8 @@ JNIEXPORT jlongArray JNICALL Java_ai_rapids_cudf_Table_filter(JNIEnv *env, jclas
   JNI_NULL_CHECK(env, mask_jcol, "mask column is null", 0);
   try {
     cudf::jni::auto_set_device(env);
-    cudf::table_view *input = reinterpret_cast<cudf::table_view *>(input_jtable);
-    cudf::column_view *mask = reinterpret_cast<cudf::column_view *>(mask_jcol);
+    auto input = reinterpret_cast<cudf::table_view *>(input_jtable);
+    auto mask = reinterpret_cast<cudf::column_view *>(mask_jcol);
```
Comment (jlowe) [inline java/src/main/native/src/TableJni.cpp:3001]:
```suggestion
    auto input = reinterpret_cast<cudf::table_view const *>(input_jtable);
    auto mask = reinterpret_cast<cudf::column_view const *>(mask_jcol);
```

File: java/src/main/native/src/TableJni.cpp [hunk 2]
```diff
@@ -3577,7 +3574,7 @@ JNIEXPORT jlongArray JNICALL Java_ai_rapids_cudf_Table_sample(JNIEnv *env, jclas
   JNI_NULL_CHECK(env, j_input, "input table is null", 0);
   try {
     cudf::jni::auto_set_device(env);
-    cudf::table_view *input = reinterpret_cast<cudf::table_view *>(j_input);
+    auto input = reinterpret_cast<cudf::table_view *>(j_input);
```
Comment (jlowe) [inline java/src/main/native/src/TableJni.cpp:3577]:
```suggestion
    auto input = reinterpret_cast<cudf::table_view const *>(j_input);
```

File: java/src/main/native/src/TableJni.cpp (hunk 1 above)
Comment (mythrocks) [reply java/src/main/native/src/TableJni.cpp]: Fixed this occurrence, and other locations modified in this PR.

File: java/src/main/native/src/TableJni.cpp (hunk 2 above)
Comment (mythrocks) [reply java/src/main/native/src/TableJni.cpp]: Fixed this occurrence, and other locations modified in this PR.

**Review by jlowe — APPROVED** [review]

**Comment by mythrocks** [comment]
One more follow-up remains: To reduce verbosity with judicious use of `auto`, mainly for the results of explicit casts. I'll fix the `nullptr` references there as well.

**Comment by mythrocks** [comment]
Resolved merge conflicts. This will be checked in after the CI completes.

**Comment by mythrocks** [comment]
Rerun tests

**Comment by mythrocks** [comment]
@gpucibot merge

**Comment by mythrocks** [comment]
This has now been merged. Thanks for the reviews, @jlowe.

I'll have to delay the remaining follow up till next week.

**Pull request merged**